server/src/main/java/org/elasticsearch/action/search/QueryPhaseResultConsumer.java

```diff
@@ -67,8 +66,8 @@ public class QueryPhaseResultConsumer extends ArraySearchPhaseResults<SearchPhas
     private final Consumer<Exception> onPartialMergeFailure;
 
     private final int batchReduceSize;
-    private final List<QuerySearchResult> buffer = new ArrayList<>();
-    private final List<SearchShard> emptyResults = new ArrayList<>();
+    private List<QuerySearchResult> buffer = new ArrayList<>();
+    private List<SearchShard> emptyResults = new ArrayList<>();
```

Comment:
Is this bit necessary? I see we were doing `clear` before, although that is not technically precisely the same thing, I think that's a different scale of problem. If needed we could make this change separately and focus on the main problem we are solving here?

Reply:
The `clear` is very complicated to understand concurrency wise and I ran into some races from it when I remove the volatile reads on the merge result.
Since it's also clearly inefficient I figured I'd just make the variable not shared and make things obviously safe. I bet I could've moved the synchronisation around to get this right, but it felt not worth it.